sdk/core/azure_core_amqp/src/cbs.rs

```diff
@@ -51,7 +51,7 @@ pub trait AmqpClaimsBasedSecurityApis {
         &self,
         path: String,
         token_type: Option<String>,
-        secret: String,
+        secret: &Secret,
```

Comment:
Why did you go from passing by value to passing by ref? Or was that unrelated?

Reply:
Because my `perform_authorization` takes an `&AccessToken` which has a `Secret` field.

I could either change the `authorize_path` function to take the secret by ref everywhere, *or* I could clone it in the refresh case.

I didn't see any value in cloning the secret value because ultimately the credential implementation going to call `Secret::secret()` which just returns the inner string slice, and thus the outer clone would be unnecessary.